libvmi/core.c

```diff
@@ -544,6 +544,20 @@ status_t vmi_init(
     }
     dbprint(VMI_DEBUG_CORE, "--completed driver init.\n");
 
+    if (init_flags & VMI_INIT_WAIT_FOR_DOMAIN) {
+        if ( VMI_FAILURE == driver_light_init(_vmi, init_flags) ) {
+            if ( error )
+                *error = VMI_INIT_ERROR_DRIVER;
+            goto error_exit;
+        }
+
+        if ( init_flags == VMI_INIT_WAIT_FOR_DOMAIN ) {
+            /* we have all we need to wait for domains. Return if there is nothing else */
+            status = VMI_SUCCESS;
+            goto error_exit;
```

Comment:
Going to error exit to return success looks wrong. You also can't just ignore all the other flags. Why are they necessarily incompatible? What if you are starting on an existing domain and just want to catch the event when it's deleted?

Reply:
I go to out only if VMI_INIT_WAIT_FOR_DOMAIN is set in the init_flags. If there are other flags then I go on

Reply:
It still looks wrong to go to error_exit with a success code.

Reply:
Agreed but the error_exit checks the status. I can move the *vmi = _vmi and return success up there but I did not want to double the code.